### PR TITLE
Make native filesystem extension work on Windows.

### PR DESCRIPTION
**Pull request by darktears**
I believe it needs to be extended and completed.

### CODE REVIEW

**Comment by darktears** [comment]
@shawngao5 PTAL

File: experimental/native_file_system/virtual_root_provider_win.cc [hunk 1]
```diff
@@ -1,12 +1,39 @@
...
+    return;
+
+  if (testing_enabled_) {
+    base::GetTempDir(&home_path_);
+  }
+  else {
+    home_path_ = base::GetHomeDir();
+  }
+
+  virtual_root_map_["DESKTOP"] = home_path_.Append(FILE_PATH_LITERAL("Desktop"));
```
Comment (pozdnyakov) [inline experimental/native_file_system/virtual_root_provider_win.cc]: "DESKTOP" (and others) are used somewhere else, right? it's worth making const variables for them

Comment (darktears) [inline experimental/native_file_system/virtual_root_provider_win.cc]: Yep in the linux provider, however this one should be rewritten to use xdg to get these names. Not sure it's worth it at the moment. I think overall the native file system access needs a lot of love.

File: experimental/native_file_system/virtual_root_provider_win.cc [hunk 2]
```diff
@@ -1,12 +1,45 @@
-// Copyright (c) 2014 Intel Corporation. All rights reserved.
```
Comment (shawngao5) [inline experimental/native_file_system/virtual_root_provider_win.cc]: 2015

File: experimental/native_file_system/virtual_root_provider_win.cc [hunk 3]
```diff
@@ -1,12 +1,45 @@
...
+  virtual_root_map_["DOWNLOADS"] =
+      home_path_.Append(FILE_PATH_LITERAL("Downloads"));
+  virtual_root_map_["DOCUMENTS"] =
+      home_path_.Append(FILE_PATH_LITERAL("Documents"));
+  virtual_root_map_["MUSIC"] =
+      home_path_.Append(FILE_PATH_LITERAL("Music"));
+  virtual_root_map_["PICTURES"] =
+      home_path_.Append(FILE_PATH_LITERAL("Pictures"));
+  virtual_root_map_["VIDEOS"] =
+      home_path_.Append(FILE_PATH_LITERAL("Videos"));
```
Comment (shawngao5) [inline experimental/native_file_system/virtual_root_provider_win.cc]: In windows such folders are called `Known Folder`, https://msdn.microsoft.com/en-us/library/windows/desktop/dd378457(v=vs.85).aspx.
These folders can be redirected to other paths than just under home directory by API, `SHSetKnownFolderPath`,  https://msdn.microsoft.com/en-us/library/windows/desktop/bb762249(v=vs.85).aspx.
So the better way for Windows is `SHGetKnownFolderPath`, https://msdn.microsoft.com/en-us/library/windows/desktop/bb762188(v=vs.85).aspx.

File: experimental/native_file_system/virtual_root_provider_win.cc [hunk 4]
```diff
@@ -1,12 +1,45 @@
...
-
-#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
-
-#include "base/logging.h"
-
-VirtualRootProvider::VirtualRootProvider() {
-  // TODO(darktears): Windows support to be added.
-  NOTIMPLEMENTED();
-}
+// Copyright (c) 2014 Intel Corporation. All rights reserved.
```
Comment (shawngao5) [inline experimental/native_file_system/virtual_root_provider_win.cc]: 2015

**Comment by darktears** [comment]
Tracked there instead : https://github.com/crosswalk-project/crosswalk/pull/2936

**Pull request closed without merging**